Floofbot/Modules/Administration.cs

```diff
@@ -146,6 +148,45 @@ public class Administration : ModuleBase<SocketCommandContext>
             await Context.Channel.SendMessageAsync("", false, builder.Build());
         }
 
+        [Command("purge")]
+        [Alias("p")]
+        [Summary("Deletes recent messages from a given user for all channels on the server")]
+        [RequireContext(ContextType.Guild)]
+        [RequireUserPermission(GuildPermission.BanMembers)]
+        public async Task PurgeUserMessages(
+            [Summary("user")] string user)
+        {
+            IUser badUser = resolveUser(user);
+            if (badUser == null) {
+                await Context.Channel.SendMessageAsync($"⚠️ Could not find user \"{user}\"");
+                return;
+            }
+
+            // retrieve user messages from ALL channels
+            foreach (ISocketMessageChannel channel in Context.Guild.TextChannels)
+            {
+                var asyncMessageCollections = channel.GetMessagesAsync(MESSAGES_TO_SCAN_PER_CHANNEL_ON_PURGE);
+                await foreach(var messageCollection in asyncMessageCollections)
+                {
+                    foreach (var message in messageCollection)
+                    {
+                        if (message.Author.Id == badUser.Id)
+                        {
+                            await channel.DeleteMessageAsync(message);
```

Comment:
could run this in a delayed async task to help with rate limits async task

Reply:
The part that needs to be delayed isn't exactly the deletion, but the logging of the messages

Reply:
delaying the deletion will delay the logging, though